<a href="https://colab.research.google.com/github/Guympele/Projets-on-Data-Analysis/blob/Data-sets-analysis/project1__Wine_Reviews_Data_Set_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!wget https://www.dropbox.com/s/wti4hu2mth7cze5/STAT%20S01V15.zip
!unzip  "STAT S01V15.zip"
import pandas as pd
import numpy as np

--2024-04-05 14:53:21--  https://www.dropbox.com/s/wti4hu2mth7cze5/STAT%20S01V15.zip
Resolving www.dropbox.com (www.dropbox.com)... 162.125.5.18, 2620:100:601d:18::a27d:512
Connecting to www.dropbox.com (www.dropbox.com)|162.125.5.18|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: /s/raw/wti4hu2mth7cze5/STAT%20S01V15.zip [following]
--2024-04-05 14:53:22--  https://www.dropbox.com/s/raw/wti4hu2mth7cze5/STAT%20S01V15.zip
Reusing existing connection to www.dropbox.com:443.
HTTP request sent, awaiting response... 302 Found
Location: https://uc29a982f79239bfd6082ab92d9a.dl.dropboxusercontent.com/cd/0/inline/CQc8HPVoO6Ekm6CxKXFC6e64Dbjq3LiZ3RhAo7uZ1XF4L5ST8qXXynZqEhvBdH0nV_Nted0fADYHblT6ITg-bHFsMu8ciSUk6NEWgilNc_lTf6u2iGXcWly75XzlZgMJQRRtkMxpK7h1a1n5DrY5IDll/file# [following]
--2024-04-05 14:53:22--  https://uc29a982f79239bfd6082ab92d9a.dl.dropboxusercontent.com/cd/0/inline/CQc8HPVoO6Ekm6CxKXFC6e64Dbjq3LiZ3RhAo7uZ1XF4L5ST8qXXynZqEhvBdH0nV_Nted0fADYHblT6ITg-bHF

In [3]:
df = pd.read_csv("winemag-data_first150k.csv")
df.head()

Unnamed: 0 country                                        description  \
0           0      US  This tremendous 100% varietal wine hails from ...   
1           1   Spain  Ripe aromas of fig, blackberry and cassis are ...   
2           2      US  Mac Watson honors the memory of a wine once ma...   
3           3      US  This spent 20 months in 30% new French oak, an...   
4           4  France  This is the top wine from La Bégude, named aft...   

                            designation  points  price        province  \
0                     Martha's Vineyard      96  235.0      California   
1  Carodorum Selección Especial Reserva      96  110.0  Northern Spain   
2         Special Selected Late Harvest      96   90.0      California   
3                               Reserve      96   65.0          Oregon   
4                            La Brûlade      95   66.0        Provence   

            region_1           region_2             variety  \
0        Napa Valley               Napa  Cabernet Sauvignon   
1               Toro                NaN       Tinta de Toro   
2     Knights Valley             Sonoma     Sauvignon Blanc   
3  Willamette Valley  Willamette Valley          Pinot Noir   
4             Bandol                NaN  Provence red blend   

                    winery  
0                    Heitz  
1  Bodega Carmen Rodríguez  
2                 Macauley  
3                    Ponzi  
4     Domaine de la Bégude

**Question** **1** : Measures of Central Tendency for - Points & Price column

In [36]:
df[['points','price']].mean() # mean values

points    87.888418
price     33.131482
dtype: float64

1. the average price of one bottle of wine is 33.131482 dollars
2. the average points receive by bottle of wine is 87.888418

In [37]:
df[['points','price']].median() # median values

points    88.0
price     24.0
dtype: float64

1. the price of the half bottles of wine was 24.0 dollars
2. the half bottle of wine receive 88.0 points

In [38]:
df[['points','price']].mode() # mode values

points  price
0      87   20.0

1. the recurrent most price of wine is 20.0 dollars
2. the recurrent most points receive by bottles of wine is 87 points

**Question 2** : Measures of spread for Points & Price

In [39]:
#Q2 Range for price column
min_price = df['price'].min()
max_price = df['price'].max()

range_price = max_price - min_price
print(range_price)

2296.0


The range price is extremely or highly spread out considering the median price

In [40]:
#Q2 Range for points column
min_point = df['points'].min()
max_point = df['points'].max()

range_point = max_point - min_point
print(range_point)

20


The range points is spread far out considering the median price

In [41]:
df[['points','price']].var(ddof=0) # population variance for points and price columns

points      10.383740
price     1319.317022
dtype: float64

the variance evaluate how far each is around the means. We observe a high value of price variance which means that values are spread far around the mean values and characterise outliers presence.

In [43]:
df[['points','price']].std(ddof=0) # population standard deviation for points and price columns

points     3.222381
price     36.322404
dtype: float64

Here we observe a high value of standard deviation for the price which means that we have too much outliers and also means values are highly far around the mean.

**Question** **3** : Measure of Shape (Skewness & Kurtosis for Points Column)

In [26]:
from scipy.stats import skew

print(f"The Skweness of Points Column is: {skew(df['points'])}")

The Skweness of Points Column is: 0.14283121423675518


the value of kweness here is greather than 0 which means that the data is left skewed.(more spread in the left side )

In [27]:
from scipy.stats import kurtosis

print(f"The Pearson Kurtosis of Points Column is:{kurtosis(df['points'], fisher=False)}")
print(f"The Fisher's kurtosis of Points Column is:{kurtosis(df['points'], fisher=True)}")

The Pearson Kurtosis of Points Column is:2.713553658335501
The Fisher's kurtosis of Points Column is:-0.2864463416644991


The value of kurtosis here means that the dataset has heavy based outliers

**Question 4** : Evaluation of the Z-score for Points Column

In [44]:
from scipy.stats import zscore

print(f"{zscore(df['points'])}")

0         2.517263
1         2.517263
2         2.517263
3         2.517263
4         2.206934
            ...   
150925    0.965616
150926    0.965616
150927    0.965616
150928    0.655286
150929    0.655286
Name: points, Length: 150930, dtype: float64


In [32]:
from sklearn.preprocessing import StandardScaler
ss = StandardScaler()
ss.fit_transform(df['points'].values.reshape(-1,1))

array([[2.51726326],
       [2.51726326],
       [2.51726326],
       ...,
       [0.96561563],
       [0.6552861 ],
       [0.6552861 ]])

The IQR for Points Column

In [35]:
from scipy.stats import iqr

print(f"The IQR is : {iqr(df['points'])}")

The IQR is : 4.0


The IQR = Q3 - Q1 is too much robust( high ) which indicate large dispersion of data a outliers presence.